sbol_utilities/helper_functions.py

```diff
@@ -80,15 +80,18 @@ def toplevel_named(doc: sbol3.Document, name: str) -> Optional[sbol3.Identified]
         raise ValueError(f'Name is not unique: {name}')
 
 
-def filter_top_level(doc: sbol3.Document, filter: Callable[[sbol3.TopLevel], bool]) -> Iterable[sbol3.TopLevel]:
+def filter_top_level(doc: sbol3.Document, filter: Callable[[sbol3.TopLevel], bool], **kwargs) -> Iterable[sbol3.TopLevel]:
     """Filters and returns iterable of TopLevel Objects in a document,
     which match a criteria set by a callable argument.
 
     :param doc: SBOL Document to search
     :param filter: Callable acting as filter on List of TopLevel objects
     :return: TopLevel iterator satisfying given filter
     """
-    return (obj for obj in doc.objects if filter(obj))
+    if kwargs:
```

Comment:
Is this if statement necessary? I think the two branches are equivalent.

Reply:
Fixed, I had thought initially that in absence of keyword arguments, kwargs would return an empty dict causing an extra arg to be passed to the helper functions.